bigquery/gcloud/aio/bigquery/bigquery.py

```diff
@@ -131,6 +133,21 @@ async def cancel(self, session: Optional[Session] = None,
         data: Dict[str, Any] = await resp.json()
         return data
 
+    async def result(self, session: Optional[Session] = None,
+                     backoff_count: int = 30) -> Dict[str, Any]:
+        for i in range(backoff_count):
+            job_data = await self.get_job(session)
+            status = job_data['status']
+            if status['state'] == 'DONE':
+                if 'errorResult' in status:
+                    raise Exception(
+                        'Job finished with errors: ', status['errors'],
+                    )
+                return job_data
+            await sleep(i * 0.5)
```

Comment:
I'm not sure we want to be setting the exact backoff parameters on behalf of our users -- everyone has different use cases and we as a library should not be enforcing our assumptions on applications.

We probably want to expose the consistently useful bits of this (eg. "check if a job is done") but allow users to do their own backoff handling however they'd like -- eg. at TalkIQ we tend to make use of the [backoff](https://pypi.org/project/backoff/) library with very specific configurations for various use-cases.

What do you think about doing something similar here? If you remove the backoff handler, this method can then be configured by clients however they'd like, eg. something like:
```python
# gcloud-aio code:
    async def result(self, session: Optional[Session] = None) -> Dict[str, Any]:
        job_data = await self.get_job(session)
        status = job_data['status']
        if status['state'] == 'DONE':
            if 'errorResult' in status:
                raise Exception('Job finished with errors: ', status['errors'])
            return job_data
        raise OSError('Job results are still pending')

# client code:
@backoff.on_exception(backoff.expo, OSError, max_tries=8)  # or whatever the client wants here!
async def get_job_result():
    return await client.result()
```

Reply:
Yeah, looks nice; Will update soon